yarn install failed

<!-- Please search existing issues to avoid creating duplicates. --> 
<!-- Also please test using the latest insiders build to make sure your issue has not already been fixed: https://code.visualstudio.com/insiders/ -->

<!-- Use Help > Report Issue to prefill these. -->
- VSCode Version: master commit a8f286bac03ad9a4659dc3d6952638173d616642
- OS Version: macOS 10.13.6

Steps to Reproduce:

1. git clone https://github.com/Microsoft/vscode.git
2. cd vscode
3. yarn

ouput: 

```bash
yarn install v1.9.4
warning package.json: No license field
warning code-oss-dev-smoke-test@0.1.0: No license field
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
$ npm run compile

> code-oss-dev-smoke-test@0.1.0 compile /Users/lyfeway/Repositories/nilcode/test/smoke
> npm run copy-driver && npm run copy-driver-definition && tsc


> code-oss-dev-smoke-test@0.1.0 copy-driver /Users/lyfeway/Repositories/nilcode/test/smoke
> cpx src/vscode/driver.js out/vscode


> code-oss-dev-smoke-test@0.1.0 copy-driver-definition /Users/lyfeway/Repositories/nilcode/test/smoke
> node tools/copy-driver-definition.js

../../../../node_modules/@types/react-router/index.d.ts(116,50): error TS2304: Cannot find name 'Exclude'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! code-oss-dev-smoke-test@0.1.0 compile: `npm run copy-driver && npm run copy-driver-definition && tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the code-oss-dev-smoke-test@0.1.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

how to fix:

update `test/smoke/package.json` typescript version from `2.5.2` to `2.9.2` will fix it.

--- a/test/smoke/package.json
+++ b/test/smoke/package.json
@@ -33,7 +33,7 @@
     "rimraf": "^2.6.1",
     "strip-json-comments": "^2.0.1",
     "tmp": "0.0.33",
-    "typescript": "2.5.2",
+    "typescript": "2.9.2",
     "watch": "^1.0.2"
   }
 }

